### PR TITLE
L2 Code Assessment

### PR DESCRIPTION
**Pull request by Manyata-Patel**

### CODE REVIEW

**Comment by AryanKammili** [comment]
You have a null pointer inside of your SwerveModule Class:

image

Right here you create a CTREconfigs object after calling a method that used the 'configs' object. The configDrive() uses the config object but when the configDrive() object is called the config object is not extantiated. So move the 'congifs' object upwards

**Comment by AryanKammili** [comment]
Make sure your current limits match the Phoenix6 documentation for current limits.

**Comment by AryanKammili** [comment]
Good job on standardizing units, make sure to add telemetry for periodic() and move it up below the constructor.

**Comment by AryanKammili** [comment]
image

Change the mods[1] to mods[i]

**Comment by Manyata-Patel** [comment]
> Make sure your current limits match the Phoenix6 documentation for current limits.

Those current limits are actually from the redux code, I took all constants such as PID values and hardware measurements from that codebase.

**Comment by AryanKammili** [comment]
image

Add 'this' to the InstantCommand for adding requirements. I think you have to do that idk kAnshul told me about it

**Comment by AryanKammili** [comment]
image


If you want to toggle between boolean buttons then you should change it from a boolean to a booleanSupplier, and use the toggle method to change it after the button click. I'm not sure about that but when you test I guess you will figure that out.

**Comment by AryanKammili** [comment]
If this code is being run on Redux make sure to run sysID later after you add the sysID file from robot-perfect-forte

**Comment by Manyata-Patel** [comment]
> image
> 
> Add 'this' to the InstantCommand for adding requirements. I think you have to do that idk kAnshul told me about it

As in do this.gyro.setYaw()?

**Comment by Manyata-Patel** [comment]
> image
> 
> If you want to toggle between boolean buttons then you should change it from a boolean to a booleanSupplier, and use the toggle method to change it after the button click. I'm not sure about that but when you test I guess you will figure that out.

Gotcha but the logic I used for this is what I did for Iron Maiden and it worked on that robot, but yeah if I face issues with it I'll change it. Also the parameter that SwerveCommand takes in is a BooleanSupplier for centric.

**Comment by Manyata-Patel** [comment]
> > image
> > Add 'this' to the InstantCommand for adding requirements. I think you have to do that idk kAnshul told me about it
> 
> As in do this.gyro.setYaw()?

For simplicity though I can just make it an instant command in RobotContainer, I'll just do that.

File: src/main/java/frc/robot/Robot.java [hunk 1]
```diff
@@ -56,11 +56,11 @@ public void disabledPeriodic() {}
   /** This autonomous runs the autonomous command selected by your {@link RobotContainer} class. */
   @Override
   public void autonomousInit() {
-    m_autonomousCommand = m_robotContainer.getAutonomousCommand();
+    Command command = RobotContainer.getAutonomousCommand();
```
Comment (mduffor) [inline src/main/java/frc/robot/Robot.java:59]: "autonomousCommand" was more descriptive.  Also, you have put the command in a local variable which leaves scope when autonomousInit() exits.  Therefore there is no place in the code where m_autonomousCommand is set, thus it won't be canceled in teleopInit().

File: src/main/java/frc/robot/subsystems/Swerve.java [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+  private boolean shouldFlip() {
+      var alliance = DriverStation.getAlliance();
+      if (alliance.isPresent()) {
+        return alliance.get() == DriverStation.Alliance.Red;
+      }
+      return false;
+    }
+  
+
+  ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
```
Comment (mduffor) [inline src/main/java/frc/robot/subsystems/Swerve.java]: 125 character columns are REALLY long... keeping lines down to 80 to 100 characters max is better.

File: src/main/java/frc/robot/RobotContainer.java [hunk 3]
```diff
@@ -5,59 +5,66 @@
...
+    swerve = new Swerve();
+    swerve.setDefaultCommand(
+      new SwerveCommand(
+        swerve, 
+        () -> driverController.getLeftY(), 
+        () -> driverController.getLeftX(), 
+        () -> driverController.getRightX(), 
+        () -> centric));
+
+    autoChooser = AutoBuilder.buildAutoChooser();
```
Comment (mduffor) [inline src/main/java/frc/robot/RobotContainer.java:51]: Do you have any control over what commands this returns?

File: src/main/java/frc/robot/commands/SwerveCommand.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  @Override
+  public void initialize() {}
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
+    double translation = translationSup.getAsDouble();
+    double strafe = strafeSup.getAsDouble();
+    double rotation = rotationSup.getAsDouble();
+    boolean centric = centricSup.getAsBoolean();
```
Comment (mduffor) [inline src/main/java/frc/robot/commands/SwerveCommand.java]: A good naming scheme for variables that are booleans is to use words like "is" to start them.  i.e.  isCentric, isCentricSup

File: src/main/java/frc/robot/commands/SwerveCommand.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+    swerve.drive(
+      // The value as of now is between 0 and 1 // 
+      // Multiplying by the max speed helps finding the real demand that was applied //
+      new Translation2d(translation, strafe).times(SwerveConstants.maxSpeed), 
+      rotation * SwerveConstants.maxAngularVelocity * 2,
```
Comment (mduffor) [inline src/main/java/frc/robot/commands/SwerveCommand.java]: Indent subsequent lines that are part of the same statement.

Comment (mduffor) [reply src/main/java/frc/robot/commands/SwerveCommand.java]: Why the magic number 2?

File: src/main/java/frc/robot/lib/DriveConversions.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+        return motorRPS;
+    }
+
+    public static double MPSToFalcon(double velocity, double circumference, double gearRatio){
+        double wheelRPM = ((velocity * 60) / circumference); // 60 seconds in a minute, divide by circumference 
+        double motorVelocity = WheelRPMToMotorRPS(wheelRPM, gearRatio);
+        return motorVelocity;
+    }
+
+    public static double toMPS(double RPS, double circumference) {
```
Comment (mduffor) [inline src/main/java/frc/robot/lib/DriveConversions.java]: Should have both javaDocs and parameter naming that indicates that the wheel circumference needs to be given in meters.

File: src/main/java/frc/robot/RobotContainer.java (hunk 3 above)
Comment (Manyata-Patel) [reply src/main/java/frc/robot/RobotContainer.java:51]: Yes, buildAutoChooser() simply returns a sendable chooser with all the PathPlannerAutos linked to this project. I've only created a simple straight path auto thus far to test.

File: src/main/java/frc/robot/commands/SwerveCommand.java (hunk 5 above)
Comment (Manyata-Patel) [reply src/main/java/frc/robot/commands/SwerveCommand.java]: So I asked Aryan about said magic number, he said he doesn't know. I'll ask him again in slack however.

**Pull request closed without merging**